File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/WoWGameData/Character/Dungeon/CharacterDungeonModel.cs
using CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.MythicKeystone;
using System.Text.Json.Serialization;

namespace CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.Dungeon;

public class CharacterDungeonModel
{
    [JsonPropertyName("character")]
    public DungeonCharacterModel Character { get; set; }

    [JsonPropertyName("expansions")]
    public DungeonExpansionModel[] Expansions { get; set; }
}

File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/wowGameData/types/dungeon/CharacterDungeonModel.ts
import type { DungeonCharacterModel } from "../mythicKeystone/DungeonCharacterModel";
import type { DungeonExpansionModel } from "./DungeonExpansionModel";

export type CharacterDungeonModel = {
    character: DungeonCharacterModel;
    expansions: DungeonExpansionModel[];
}